src/js/common/schemaform/helpers.js

```diff
@@ -318,7 +318,7 @@ export function updateRequiredFields(schema, uiSchema, formData) {
     const newRequired = Object.keys(schema.properties).reduce((requiredArray, nextProp) => {
       const field = uiSchema[nextProp];
       if (field && field['ui:required']) {
-        const isRequired = field['ui:required'](formData);
+        const isRequired = field['ui:required'](pageData, formData);
```

Comment:
On the other hand, all the data in `pageData` will also be in `formData`; we may not _need_ all the rest of it, but if we do, we wouldn't need to grab two variables when we only need one.

Either way, it's not a big deal.